Fix indexer multi-languages for release/push

kephasdev opened the pull request:
#### Description
Use the indexer default languages configuration for multi-languages when a release with `MULTI` is pushed using the API `/release/push`.
The Interactive search and RSS sync seems to be already correct.

I had to change `IndexerBase.cs` to put the regex expression for MULTI in `FileNameBuilder.cs`, not quite sure if it's the right place.

Note : this is my first PR for Sonarr, I tried to look around at the function, tests naming convention to use it. If something is not good enough for a merge, feel free to say it I and will change it.

#### Issues Fixed or Closed by this PR
* Closes #7059

markus101 commented on src/NzbDrone.Core/Organizer/FileNameBuilder.cs:
```diff
@@ -55,6 +55,8 @@ public class FileNameBuilder : IBuildFileNames
         public static readonly Regex SeasonRegex = new Regex(@"(?<season>\{season(?:\:0+)?})",
                                                               RegexOptions.Compiled | RegexOptions.IgnoreCase);
 
+        public static readonly Regex MultiRegex = new (@"[_. ](?<multi>multi)[_. ]", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
This should go in `Parser`, since it is for parsing, not renaming files.

kephasdev replied:
Ok, I've moved it to `Parser` and add a new function `IsMultiLanguages` as all other RegEx were private.

markus101 commented on src/NzbDrone.Core/Indexers/IndexerBase.cs:
```diff
@@ -94,7 +92,7 @@ protected virtual IList<ReleaseInfo> CleanupReleases(IEnumerable<ReleaseInfo> re
             {
                 // Use multi languages from setting if ReleaseInfo languages is empty
-                if (c.Languages.Empty() && MultiRegex.IsMatch(c.Title) && settings.MultiLanguages.Any())
+                if (c.Languages.Empty() && FileNameBuilder.MultiRegex.IsMatch(c.Title) && settings.MultiLanguages.Any())
```
If this is done in `AggregateLanguages` does it still need to be done here too?

kephasdev replied:
Not sure what you mean here.. this function is called from different places, RSS Sync, Interactive searches, release/push

markus101 commented on src/NzbDrone.Core/Download/Aggregation/Aggregators/AggregateLanguages.cs:
```diff
@@ -11,10 +13,13 @@ namespace NzbDrone.Core.Download.Aggregation.Aggregators
+        private readonly IIndexerFactory _indexerFactory;
 
-        public AggregateLanguages(Logger logger)
+        public AggregateLanguages(Logger logger,
```
Logger should be last (and reorder other lines to match)

markus101 commented:
Just some comment after a quick pass.

kephasdev commented:
> Just some comment after a quick pass.

Not sure what is the prefered method for Sonarr, I updated the PR and squashed the changes to keep it at 1 commit.

kephasdev commented on src/NzbDrone.Core/Download/Aggregation/Aggregators/AggregateLanguages.cs:
```diff
@@ -71,6 +75,17 @@ public RemoteEpisode Aggregate(RemoteEpisode remoteEpisode)
+            {
+                var indexer = _indexerFactory.Get(releaseInfo.IndexerId);
+
+                if (indexer?.Settings is IIndexerSettings settings && releaseInfo.Title != null && Parser.Parser.IsMultiLanguages(releaseInfo.Title) && settings.MultiLanguages.Any())
```
Done for the previous comments too

kephasdev commented on src/NzbDrone.Core/Indexers/IndexerBase.cs:
```diff
@@ -94,7 +91,7 @@ protected virtual IList<ReleaseInfo> CleanupReleases(IEnumerable<ReleaseInfo> re
             {
                 // Use multi languages from setting if ReleaseInfo languages is empty
-                if (c.Languages.Empty() && MultiRegex.IsMatch(c.Title) && settings.MultiLanguages.Any())
+                if (c.Languages.Empty() && Parser.Parser.IsMultiLanguages(c.Title) && settings.MultiLanguages.Any())
```
Done too

markus101 commented on src/NzbDrone.Core/Parser/Parser.cs:
```diff
@@ -959,6 +961,11 @@ public static string RemoveFileExtension(string title)
             return title;
         }
 
+        public static bool IsMultiLanguages(string title)
```
The naming of this mixes singular and plural casing, `HasMultipleLanguages` or `IsMultiLanguage`  would be more appropriate.

markus101 reviewed (approved):
Thanks for all the fixes!
